sycl/doc/extensions/supported/sycl_ext_intel_device_info.md

```diff
@@ -489,6 +490,144 @@ Then the memory bus width can be obtained using the standard get\_info() interfa
       auto MemoryBusWidth = dev.get_info<ext::intel::info::device::memory_bus_width>();
     }
 
+# Throttle reason #
+
+A new device descriptor is added which provides the current GPU clock throttle reasons.
+A new enum is added with the list of possible throttle reasons.
+
+## Version ##
+
+The extension supports this query in version 7 and later.
+
+## Throttle reasons ##
+
+| Reason             | Description |
+| ------------------ | ----------- |
+| `power_cap` | The GPU clock frequency is throttled due to hitting the power limit. |
+| `current_limit` | The GPU clock frequency is throttled due to hitting the current limit. |
+| `thermal_limit` | The GPU clock frequency is throttled due to hitting the thermal limit. |
+| `psu_alert` | The GPU clock frequency is throttled due to power supply assertion. |
+| `sw_range` | The GPU clock frequency is throttled due to software supplied frequency range. |
+| `hw_range` | The GPU clock frequency is throttled because there is a sub block that has a lower frequency when it receives clocks. |
+| `other` | The GPU clock frequency is throttled due to other reason. |
+
+
+```
+namespace sycl::ext::intel {
+
+  enum class throttle_reason {
+    power_cap,
+    current_limit,
+    thermal_limit,
+    psu_alert,
+    sw_range,
+    hw_range,
+    other
+  }
+
+}
+```
+
+## Device Information Descriptors ##
+
+| Device Descriptors | Return Type | Description |
+| ------------------ | ----------- | ----------- |
+| `ext::intel::info::device::current_clock_throttle_reasons` | `std::vector<ext::intel::throttle_reason>` | Returns the set of throttle reasons describing why the frequency is being limited by the hardware. Returns empty set if frequency is not throttled. |
```

Comment:
As written, this descriptor is GPU-specific: the introduction talks about the "GPU clock" and the descriptions of the throttle reasons say "GPU".  The other queries that are GPU-specific (e.g. number of EUs) are all prefixed with `gpu_`.

I think we should either:

1. Rename this to `gpu_current_throttle_reasons`; or
2. Remove "GPU" from the introduction and description.

I have a slight preference for 2, because the throttle reasons are already guarded by an aspect.  If we never implement this extension on another device (e.g., CPU), we just won't enable the aspect anywhere except GPUs.  But removing "GPU" from the description gives us the option to try and implement it on other devices.

Reply:
Thank you, sounds reasonable, I have applied option 2.